src/idl_gen_cpp.cpp

```diff
@@ -591,6 +591,20 @@ class CppGenerator : public BaseGenerator {
                       "std::string";  // Only for custom string types.
   }
 
+  bool NativeVectorDefined(const FieldDef *field) {
+    auto attr = field ? field->attributes.Lookup("cpp_vec_type") : nullptr;
+    auto &ret = attr ? attr->constant : parser_.opts.cpp_object_api_vector_type;
+    if (ret.empty()) { return false; }
+    return true;
+  }
+
+  const std::string NativeVector(const FieldDef *field) {
+    auto attr = field ? field->attributes.Lookup("cpp_vec_type") : nullptr;
+    auto &ret = attr ? attr->constant : parser_.opts.cpp_object_api_vector_type;
```

Comment:
this does kinda the same. Maybe a better way to factor this is one function that does Field->std::string * for the vector type, which returns null for not present, that way this logic can be shared.

Reply:
Yeah true, the lookup code for ptr, string and vector type is the same, I've abstracted that away into FieldStringLookup().